integration/util/web.go

```
@@ -137,11 +138,32 @@ func SubscribeForWSMessages(cfg *TestConfiguration, conn *websocket.Conn, eventT
 	return WaitForWSMessage(cfg, conn, fmt.Sprintf("%s:ACK", eventType))
 }
 
+func logError(t *testing.T, f func() error, description string) {
+	if err := f(); err != nil {
+		t.Logf("error calling %s: %v", description, err)
+	}
+}
+
+func unsubscribe(cfg *TestConfiguration, ws *websocket.Conn, messageType string) error {
+	if err := SubscribeForWSMessages(cfg, ws, messageType, ""); err != nil {
+		return fmt.Errorf("unable to unsubscribe with %s: %v", messageType, err)
+	}
+	return nil
+}
+
+// UnsubscribeFromWSMessages unsubscribes for the messages that are sent from a web socket session
+// and awaits confirmation response.
+// It logs any errors that occur, so that it can be called with defer.
+func UnsubscribeFromWSMessages(t *testing.T, cfg *TestConfiguration, ws *websocket.Conn, messageType string) {
```

Maybe we should return the error, not log it

---

This is so we can call this with defer:

```
	ws, err := util.NewDigitalTwinWSConnection(suite.Cfg)
	require.NoError(suite.T(), err, "cannot create a websocket connection to the backend")
	defer Disconnect(suite.T(), suite.Cfg, ws)

	filter := fmt.Sprintf("like(resource:path,'/features/%s/*')", featureID)
	err = util.SubscribeForWSMessages(suite.Cfg, ws, "START-SEND-EVENTS", filter)
	require.NoError(suite.T(), err, "subscription for events should succeed")
	defer Unsubscribe(suite.T(), suite.Cfg, ws, "STOP-SEND-EVENTS")
```

---

Callers may choose to ignore errors returned and defer will work even if a method returns an error.